shell/browser/native_window_mac.mm

```diff
@@ -792,10 +796,29 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   return [window_ isMiniaturized];
 }
 
+void NativeWindowMac::HandlePendingFullscreenTransitions() {
+  if (!pending_transitions_.size())
+    return;
+
+  const auto& next_transition = pending_transitions_.front();
+  pending_transitions_.erase(pending_transitions_.begin());
+
+  SetFullScreen(next_transition);
+}
+
 void NativeWindowMac::SetFullScreen(bool fullscreen) {
   if (fullscreen == IsFullscreen())
```

Comment:
This check should be modified to test the tail of `pending_transitions_` instead of current fullscreen state. Otherwise following calls would not work:

```is
// current state: not in fullscreen
window.setFullScreen(true);  // current state: fullscreen
window.setFullScreen(false); // pushed. current state: fullscreen
window.setFullScreen(true);  // ignored as current state is fullscreen
```